aws/resource_aws_kinesis_firehose_delivery_stream.go

```diff
@@ -41,6 +41,59 @@ func cloudWatchLoggingOptionsSchema() *schema.Schema {
 	}
 }
 
+func S3BackupConfigurationSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeSet,
```

Comment:
@radeksimko I have a doubt here. When do we use `TypeSet` vs `TypeList`?
I see `S3ConfigurationSchema` has `TypeList` with `MaxItems: 1`. Is this significant only during type assertion?

Reply:
In other words it would be better to use `TypeList` here.

Can't we call this `s3ConfigurationSchema` and use it in `"s3_configuration"` too to reduce the repetition a bit?

Reply:
>Can't we call this s3ConfigurationSchema and use it in "s3_configuration" too to reduce the repetition a bit?

That's what I was going for and noticed that the schemaTypes are different.

Reply:
> Set is used when order doesn't matter

@radeksimko Just curious. Why do we use `TypeList` here then? The order doesn't matter here, right?

Reply:
Correct, it doesn't, but also there's really nothing to order 😃  as we only ever have a single item in the list and `TypeList` makes it more convenient for referencing purposes, i.e. `field_name.0.nested_field` instead of `field_name.<computed-index>.nested_field`.